pkg/scheduler/core/generic_scheduler.go

```diff
@@ -203,27 +205,60 @@ func (g *genericScheduler) findNodesThatFitPod(ctx context.Context, fwk framewor
 
 	// Run "prefilter" plugins.
 	s := fwk.RunPreFilterPlugins(ctx, state, pod)
+	allNodes, err := g.nodeInfoSnapshot.NodeInfos().List()
+	if err != nil {
+		return nil, nil, err
+	}
 	if !s.IsSuccess() {
 		if !s.IsUnschedulable() {
 			return nil, nil, s.AsError()
 		}
 		// All nodes will have the same status. Some non trivial refactoring is
 		// needed to avoid this copy.
-		allNodes, err := g.nodeInfoSnapshot.NodeInfos().List()
-		if err != nil {
-			return nil, nil, err
-		}
 		for _, n := range allNodes {
 			filteredNodesStatuses[n.Node().Name] = s
 		}
 		return nil, filteredNodesStatuses, nil
 	}
 
-	feasibleNodes, err := g.findNodesThatPassFilters(ctx, fwk, state, pod, filteredNodesStatuses)
+	var feasibleNodes []*v1.Node
+
+	// The "NominatedNodeName" field is set due to the preemption in the previous scheduling cycle, the node
+	// most likely is still the only candidate for the preemptor pod.
+	// Have a try on this node directly when the field is set and the feature is enabled.
+	preferNominatedNode := len(pod.Status.NominatedNodeName) > 0 && utilfeature.DefaultFeatureGate.Enabled(features.PreferNominatedNode)
+	if preferNominatedNode {
+		nnn := pod.Status.NominatedNodeName
+		nodeInfo, err := g.nodeInfoSnapshot.Get(nnn)
+		if err != nil {
+			return nil, nil, err
+		}
+		node := []*framework.NodeInfo{nodeInfo}
+		nominatedNode, err := g.findNodesThatPassFilters(ctx, fwk, state, pod, filteredNodesStatuses, node)
+		if err != nil {
+			return nil, nil, err
+		}
+		if len(nominatedNode) != 0 {
```

Comment:
doesn't findNodesThatPassExtenders handle this? we don't have the same check when we filter the rest of the pods.

Reply:
Sorry,  I don't quite understand this, do you mean "the rest of the nodes"? Shouldn't the same check has been done at line 262?